packages/react-components/source/react/library/sidepanel/SidePanel.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '../button';
+import Heading from '../heading';
+
+const propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.node,
+  contentClassName: PropTypes.string,
+  onClose: PropTypes.func,
+  open: PropTypes.bool,
+  title: PropTypes.string,
+  type: PropTypes.oneOf(['toolbar']).isRequired,
+};
+
+const defaultProps = {
+  children: null,
+  open: true,
+  title: '',
+  onClose() {},
+  className: '',
+  contentClassName: '',
+};
+
+/**
+ * SidePanel may be used inside the rightmost Columns.Column.
+ */
+const SidePanel = ({
+  children,
+  open,
+  title,
+  onClose,
+  className,
+  contentClassName,
+}) => {
+  const actions = (
+    <Button
+      icon="close"
+      type="transparent"
+      innerFocus
+      aria-label="Close side panel"
+      onClick={onClose}
+    />
```

Comment:
We need to add back spreading props on this `<Button>` like we do in nebula-ui's [Sidebar.jsx#L58](https://github.com/puppetlabs/nebula-ui/blob/e787dd756f552648c7e6cf7ae134c6a7de35065c/src/client/components/Sidebar/Sidebar.jsx#L58) to enable React Router links, which will be required once https://github.com/puppetlabs/nebula-ui/pull/411 lands.

Reply:
I merged that PR this morning FYI. Instead of spreading all props maybe we could add a `closeButtonProps` prop on `sidePanel` so that it's clear what they're being applied to?